x-pack/plugins/ml/public/application/timeseriesexplorer/components/timeseries_chart/timeseries_chart.js

```diff
@@ -766,15 +781,53 @@ class TimeseriesChartIntl extends Component {
         )
       );
 
+    const that = this;
+
     // Remove dots that are no longer needed i.e. if number of chart points has decreased.
     dots.exit().remove();
     // Create any new dots that are needed i.e. if number of chart points has increased.
     dots
```

Comment:
Did you look at adding the popover to the multi bucket markers too? I think you should be able to use exactly the same logic.

Reply:
Added support for multi bucket anomalies in [b246b80](https://github.com/elastic/kibana/pull/175556/commits/b246b80affd874467326a98dfdbb4247e6467ed5).